x/genutil/genaccounts.go

```diff
@@ -15,133 +15,155 @@ import (
 	genutiltypes "github.com/cosmos/cosmos-sdk/x/genutil/types"
 )
 
-// AddGenesisAccount adds a genesis account to the genesis state.
+type GenesisAccount struct {
+	// Base
+	Address string    `json:"address"`
+	Coins   sdk.Coins `json:"coins"`
+
+	// Vesting
+	VestingAmt   sdk.Coins `json:"vesting_amt,omitempty"`
+	VestingStart int64     `json:"vesting_start,omitempty"`
+	VestingEnd   int64     `json:"vesting_end,omitempty"`
+
+	// Module
+	ModuleName string `json:"module_name,omitempty"`
+}
+
+// AddGenesisAccounts adds a genesis account to the genesis state.
 // Where `cdc` is client codec, `genesisFileUrl` is the path/url of current genesis file,
 // `accAddr` is the address to be added to the genesis state, `amountStr` is the list of initial coins
 // to be added for the account, `appendAcct` updates the account if already exists.
 // `vestingStart, vestingEnd and vestingAmtStr` respectively are the schedule start time, end time (unix epoch)
 // `moduleName` is the module name for which the account is being created
 // and coins to be appended to the account already in the genesis.json file.
-func AddGenesisAccount(
+func AddGenesisAccounts(
```

Comment:
As it's api break in the 0.50 backport we will keep both around.
On main + 0.52 we can do this indeed.

Reply:
This would be It’s own function in the backport, leaving the original non API breaking. I can also make a dedicated manual backport when it comes time

This is just the proper way for both to simplify future maintenance the proper way.